secant/Util/MnemonicSeedPhraseProvider.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  MnemonicSeedPhraseProvider.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 03/09/2022.
+//
+
+import Foundation
+import MnemonicSwift
+
+struct MnemonicSeedPhraseProvider {
+    /// Random 24 words mnemonic phrase
+    var randomMnemonic: () throws -> String
+    /// Random 24 words mnemonic phrase as array of words
+    var randomMnemonicWords: () throws -> [String]
+    /// Generate deterministic seed from mnemonic phrase
+    var toSeed: (String) throws -> [UInt8]
+    /// Get this mnemonic phrase as array of words
+    var asWords: (String) throws -> [String]
+    /// Validates whether the given mnemonic is correct
+    var isValid: (String) throws -> Void
+}
+
+extension MnemonicSeedPhraseProvider {
+    static let live = MnemonicSeedPhraseProvider(
+        randomMnemonic: {
+            try Mnemonic.generateMnemonic(strength: 256)
+        },
+        randomMnemonicWords: {
+            try Mnemonic.generateMnemonic(strength: 256).components(separatedBy: " ")
+        },
+        toSeed: { mnemonic in
+            let data = try Mnemonic.deterministicSeedBytes(from: mnemonic)
+
+            return [UInt8](data)
+        },
+        asWords: { mnemonic in
+            mnemonic.components(separatedBy: " ")
+        },
+        isValid: { mnemonic in
+            try Mnemonic.validate(mnemonic: mnemonic)
+        }
+    )
+    
+    static let mock = MnemonicSeedPhraseProvider(
+        randomMnemonic: {
+            """
+            still champion voice habit trend flight \
+            survey between bitter process artefact blind \
+            carbon truly provide dizzy crush flush  \
```

Comment:
I think this line has a double space and will (and probably should) break separating the string by components using space as a separator

Reply:
fixed